src/Font/UiFontDescriptor.cc

```diff
@@ -0,0 +1,57 @@
+#include "../../ui.h"
+#include "../ui-node.h"
+#include "nbind/nbind.h"
+
+UiFontDescriptor::UiFontDescriptor(uiFontDescriptor * desc) {
+	d = desc;
+	cleanup = 1;
+}
+
+UiFontDescriptor::UiFontDescriptor(char *family, double size, int weight, int italic, int stretch) {
+    d = new uiFontDescriptor();
+    d->Family = family;
+    d->Size = size;
+    d->Weight = weight;
+    d->Italic = italic;
+    d->Stretch = stretch;
+}
+
+void UiFontDescriptor::free() {
+	if(cleanup){
+		uiFreeFontButtonFont(d);
+	}
+	delete d;
```

Comment:
libui example contains a specific way to deallocate a font descritptor: `uiFreeFontButtonFont `, I think we should use it:

https://github.com/andlabs/libui/blob/master/examples/drawtext/main.c#L109

Reply:
Documentation for `uiFreeFontButtonFont`:
> uiFreeFontButtonFont() frees resources **allocated in desc by uiFontButtonFont()**.
After calling uiFreeFontButtonFont(), the contents of desc should be assumed to be undefined (though since you allocate desc itself, you can safely reuse desc for other font descriptors).
**Calling uiFreeFontButtonFont() on a uiFontDescriptor not returned by uiFontButtonFont() results in undefined behavior.**

Reply:
Ah ok, so I guess cleanup flag is used to discern if `uiFontDescriptor` is created in the constructor or passed as argument?

Beware that when cleanup===true, you are freeing `d` two times.

Reply:
Am I? `uiFreeFontButtonFont` frees only the members of FontDescriptor:

```cpp
void uiFreeFontButtonFont(uiFontDescriptor *desc)
{
	uiFreeText((char *) (desc->Family));
}
```

Reply:
Oh that's ok. Seems a little unintuitive on libui side... But maybe it's just me...